src/Web/Controllers/AccountController.cs

```diff
@@ -73,14 +81,48 @@ public async Task<IActionResult> SignIn(LoginViewModel model, string returnUrl =
             }
             ViewData["ReturnUrl"] = returnUrl;
 
+            ApplicationUser applicationUser = new ApplicationUser
+            {
+                UserName = model.Email
+            };
+
+            var signIn = new SignIn
+            {
+                SignInId = Guid.NewGuid().ToString(),
+                PasswordHash = _userManager.PasswordHasher.HashPassword(applicationUser, model.Password),
+                MerchantLocalDate = DateTimeOffset.Now,
+                CustomerLocalDate = model.DeviceFingerPrinting.ClientDate,
+                UserId = model.Email,
+                DeviceContextId = model.DeviceFingerPrinting.SessionId,
+                AssessmentType = AssessmentType.Protect.ToString(),
+                CurrentIpAddress = _contextAccessor.HttpContext.Connection.RemoteIpAddress.ToString()
+            };
+
+            var signInResponse = await _fraudProtectionService.PostSignIn(signIn);
+
+            var fraudProtectionIO = new FraudProtectionIOModel(signIn, signInResponse, "SignIn");
+            TempData.Put(FraudProtectionIOModel.TempDataKey, fraudProtectionIO);
+
+
             var result = await _signInManager.PasswordSignInAsync(model.Email, model.Password, model.RememberMe, lockoutOnFailure: false);
-            if (result.Succeeded)
+            if (!result.Succeeded)
             {
-                await TransferBasketToEmailAsync(model.Email);
-                return RedirectToLocal(returnUrl);
+                ModelState.AddModelError(string.Empty, "Invalid login attempt.");
+                return View(model);
             }
-            ModelState.AddModelError(string.Empty, "Invalid login attempt.");
-            return View(model);
+
+            //2 out of 3 signIn will be successful
+            var rejectSignIn = new Random().Next(0, 3) != 0;
+            if(rejectSignIn)
```

Comment:
can you ctrl+k+d this file too... e.g. there should be a space after the if before the opening parens.